res/controllers/midi-components-0.0.js

```diff
@@ -733,16 +733,53 @@
     });
 
     const JogWheelBasic = function(options) {
+        if (options.deck !== undefined && options.group !== undefined) {
+            console.warn(
+                "options.deck and option.group are both set; " +
+                "options.deck will take priority"
+            );
+        }
+
+        const deck = options.deck;
+        const group = script.deckFromGroup(options.group);
+        delete options.deck;
+        delete options.group;
+
         Component.call(this, options);
 
-        if (!Number.isInteger(this.deck)) {
-            console.warn("missing scratch deck");
-            return;
-        }
-        if (this.deck <= 0) {
-            console.warn("invalid deck number: " + this.deck);
-            return;
+        this._deck = undefined;
+        this._group = undefined;
+
+        Object.defineProperties(this, {
+            "deck": {
+                get: function() { return this._deck; }.bind(this),
+                set: function(value) {
+                    if (Number.isInteger(value) && value > 0) {
+                        this._deck = value;
+                        this._group = `[Channel${value}]`;
+                        this.reset();
+                    }
+                }.bind(this),
+            },
+            "group": {
+                get: function() { return this._group; }.bind(this),
```

Comment:
I think the `.bind(this)` is unnecessary. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#get

Reply:
Yeah, I wasn't really sure since my IDE still warned me about a potential bad usage of `this`. I can remove.

Reply:
yeah, or maybe just replace with an arrow function outright? That makes it clear where `this` comes from.